test/test.cc

```diff
@@ -1221,19 +1221,17 @@ void TestOutputStages(int rows, int depth, int cols, int result_offset,
       &context, lhs.const_map(), rhs.const_map(), &result_quantized_down_int32,
       lhs_offset, rhs_offset, quantize_down_pipeline);
 
-  std::uint64_t sum = 0;
```

Comment:
It was a bug --- but the bug was not hit, because the lhs_offset, rhs_offset hardcoded here are not large-negative enough to actually produce negative output values; consequently, we just never had negative values here, and so the bug was not triggered. In the final form of this PR , it also is not triggered, but an intermediate stage of this PR was attempting to exercise negative values, and that triggered it.